### PR TITLE
Some command

### PR DESCRIPTION
**Pull request by Owllyy**
No need to worry, everything is hardcode, trust me it works.

### CODE REVIEW

**Review by MauriceNDS — APPROVED** [review]
GJ

File: includes/modules/api/Channel.hpp [hunk 1]
```diff
@@ -2,28 +2,68 @@
...
-    vector<User *> users;
+	string topic;
+    set<User *> users;
+	set<User *> chanop;
 
 public:
 	Channel(const string& name) : name(name) {}
 
-	const vector<User *>& getUsers() const {
+	const set<User *>& getUsers() {
```
Comment (SimonCROS) [inline includes/modules/api/Channel.hpp:22]:
```suggestion
	const set<User *>& getUsers() const {
```

File: includes/modules/api/Channel.hpp [hunk 2]
```diff
@@ -2,28 +2,68 @@
...
+		topic = arg;
+	}
+	
 	const string& getName() const {
 		return name;
 	}
 
 	void send(const string& message) const {
-		for (vector<User *>::const_iterator it = users.begin(); it != users.end(); it++)
+		for (set<User *>::iterator it = users.begin(); it != users.end(); it++)
```
Comment (SimonCROS) [inline includes/modules/api/Channel.hpp]:
```suggestion
		for (set<User *>::const_iterator it = users.begin(); it != users.end(); it++)
```

File: modules/core/Irc.cpp [hunk 3]
```diff
@@ -45,6 +45,46 @@ void Irc::start() {
...
+		.build()
+	);
+	commandManager.registerCommand(CommandSpec::Builder()
+		.name("PRIVMSG")
+		.argument("target", new MsgToCommandElement())
+		.argument("message", GenericArguments::string())
+		.middleware(new RegisteredUserMiddleware())
+		.executor(new TestCommand())
+		.build()
+	);
```
Comment (SimonCROS) [inline modules/core/Irc.cpp]: Duplicated command, need to patch `DuplicatedCommandException`.

File: modules/core/Irc.cpp [hunk 4]
```diff
@@ -45,6 +45,46 @@ void Irc::start() {
...
+		.name("TOPIC")
+		.argument("channel", new ChannelCommandElement(false))
+		.argument("topic", GenericArguments::optional(GenericArguments::string()))
+		.middleware(new RegisteredUserMiddleware())
+		.executor(new TestCommand())
+		.build()
+	);
+	commandManager.registerCommand(CommandSpec::Builder()
+		.name("PRIVMSG")
+		.argument("target", new MsgToCommandElement())
```
Comment (SimonCROS) [inline modules/core/Irc.cpp]: Change `target` to `msgtarget` and set `MsgToCommandElement` to a list.

File: modules/core/Irc.cpp [hunk 5]
```diff
@@ -45,6 +45,46 @@ void Irc::start() {
...
+		.argument("message", GenericArguments::string())
```
Comment (SimonCROS) [inline modules/core/Irc.cpp]: Change the not provided message using `CommandElement::Builder()`.

**Review by SimonCROS — CHANGES_REQUESTED** [review]

**Review by SimonCROS — APPROVED** [review]

**Pull request merged**